src/lxd.py

```diff
@@ -0,0 +1,321 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""LXD client interface.
+
+The Lxd class is intend to be layer of abstraction to isolate the underlying implementation of LXD.
+"""
+from __future__ import annotations
+
+import io
+import tempfile
+from subprocess import CalledProcessError  # nosec B404
```

Comment:
For these and any other disables we have a standard for how to document disables: https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#static-code-analysis, is it applicable here?

Reply:
For imports, from my testing the formatter with insert a blank line above the comments in the import section. This will mess up the import sections.

I think a while back in a PR for GitHub runner charm. This came up in a PR, I remember I decide not comment on the import and rather comment when the import is used.
This has the advantage of:
- Not messing up the formatter for the import section.
- Justify each instance of the usage, rather then the whole file.

Should I update the contributing guideline with this?

Reply:
Yes let's update the contributing guide